### ISSUE TITLE
MNIST Tutorial Fails

### ISSUE THREAD

**Issue by rickhg12hs** [malmaud/TensorFlow.jl#352]
Both the [softmax regression model](https://malmaud.github.io/TensorFlow.jl/latest/tutorial.html#Building-a-softmax-regression-model-1) and the [multi-layer convolutional network](https://malmaud.github.io/TensorFlow.jl/latest/tutorial.html#Build-a-multi-layer-convolutional-network-1) fail with accuracies of `0.0`.

Interestingly, both the [Basic usage](https://malmaud.github.io/TensorFlow.jl/latest/basic_usage.html#Basic-usage-1) and [Logistic regression](https://malmaud.github.io/TensorFlow.jl/latest/logistic.html#Logistic-regression-1) examples seem to work fine.

```julia
julia> versioninfo()
Julia Version 0.6.1
Commit 0d7248e2ff (2017-10-24 22:15 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, skylake)
julia> Pkg.installed("TensorFlow")
v"0.7.5"
julia> tf_version()
v"1.4.0"
```

**Comment by tkluck** [comment]
I ran into a similar issue, and traced it down to the following. The tutorial gives this line of code:

    julia> correct_prediction = indmax(y,2) .== indmax(y_, 2)
    false

But as you can see, this doesn't return a `Tensor` but the constant value `false`. The intended line seems to have been:

    julia> correct_prediction = equal(indmax(y,2), indmax(y_, 2))
    <Tensor Equal:1 shape=unknown dtype=Bool>

and indeed, this allows me to reproduce the stated accuracy for the softmax model.

Recent changes to `.` and `broadcast` may be to blame, but I haven't dug into that (yet).

Thoughts?

**Comment by malmaud** [comment]
Ah ya, that's possible. A PR fixing the documentation would be welcome.

**Comment by rickhg12hs** [comment]
> I ran into a similar issue, and traced it down to the following. The tutorial gives this line of code:
> 
> julia> correct_prediction = indmax(y,2) .== indmax(y_, 2)
> false
> But as you can see, this doesn't return a Tensor but the constant value false. The intended line seems to have been:
> 
> julia> correct_prediction = equal(indmax(y,2), indmax(y_, 2))
> <Tensor Equal:1 shape=unknown dtype=Bool>
> and indeed, this allows me to reproduce the stated accuracy for the softmax model.
> 
> Recent changes to . and broadcast may be to blame, but I haven't dug into that (yet).
> 
> Thoughts?

Or is the true intention to overload `==`? ... Or something similar?

**Comment by rickhg12hs** [comment]
Same issue reported at #364 .
